easygcm/src/main/java/eu/inloop/easygcm/GcmHelper.java

```diff
@@ -29,9 +26,11 @@
     private GcmListener mGcmListener;
     private GcmServicesHandler mCheckServicesHandler;
     static volatile boolean sLoggingEnabled = true;
+
     private final AtomicBoolean mRegistrationRunning = new AtomicBoolean(false);
-    private static final int DEFAULT_BACKOFF_MS = 2000;
-    private static final int MAX_RETRIES = 5;
+    private RegistrationListener mRegistrationListener;
+
+    private Set<String> mSenderIdsToRegister = new HashSet<>();
```

Comment:
This is my misunderstanding and wrong implementation. There is ever going to be a single senderId - the one defined in google-services.json. It is not needed to persist it between app kills (the registration should run again). A simple inmemory boolean value indicating previous gcm registration attempt should suffice.